core/container.go

```diff
@@ -249,7 +249,13 @@ func (container *Container) From(ctx context.Context, gw bkgw.Client, addr strin
 		return nil, err
 	}
 
-	return ctr.UpdateImageConfig(ctx, func(specs.ImageConfig) specs.ImageConfig {
+	return ctr.UpdateImageConfig(ctx, func(config specs.ImageConfig) specs.ImageConfig {
+		// merge config.Env with imgSpec.Config.Env
+		newEnv := []string{}
+		if config.Env != nil {
+			newEnv = append(config.Env, imgSpec.Config.Env...)
```

Comment:
I think the linter complaint here is legit, this may result in a shared slice getting overwritten with different values. Can just initialize newEnv to be a copy of config.Env and then append to that.

Not 100% sure but this may be the culprit of the strange looking test failures.

Reply:
Thanks @sipsma for the review. Updated code with the fix.